File: inform/Settings/Specifics/IFAdvancedSettings.m
//
//  IFAdvancedSettings.m
//  Inform
//
//  Created by Andrew Hunter on 10/10/2004.
//  Copyright 2004 Andrew Hunter. All rights reserved.
//

#import "IFAdvancedSettings.h"
#import "IFUtility.h"
#import "IFCompilerSettings.h"
@implementation IFAdvancedSettings {
    IBOutlet NSButton* allowLegacyExtensionDirectory;
}

- (instancetype) init {
	return [self initWithNibName: @"AdvancedSettings"];
}

- (NSString*) title {
	return [IFUtility localizedString: @"Extensions Settings"];
}

#pragma mark - Setting up

- (void) updateFromCompilerSettings {
    IFCompilerSettings* settings = [self compilerSettings];
	
    [allowLegacyExtensionDirectory setState: [settings allowLegacyExtensionDirectory]?NSControlStateValueOn:NSControlStateValueOff];
}

- (void) setSettings {
    IFCompilerSettings* settings = [self compilerSettings];

	[settings setAllowLegacyExtensionDirectory: [allowLegacyExtensionDirectory state]==NSControlStateValueOn];
}

- (BOOL) enableForCompiler: (NSString*) compiler {
	return YES;
}

@end

File: inform/Settings/Specifics/IFAdvancedSettings.swift
//
//  IFAdvancedSettings.swift
//  Inform
//
//  Created by Toby Nelson on 28/09/2023.
//

import Foundation

class IFAdvancedSettings: IFSetting {
    @IBOutlet var allowLegacyExtensionDirectory: NSButton!

    convenience init() {
        self.init(nibName:"AdvancedSettings")
    }

    override var title: String {
        return IFUtility.localizedString("Extensions Settings")
    }

    override func updateFromCompilerSettings() {
        let settings = self.compilerSettings!

        allowLegacyExtensionDirectory.state = settings.allowLegacyExtensionDirectory ? NSControl.StateValue.on : NSControl.StateValue.off
    }

    override func setSettings() {
        let settings = self.compilerSettings

        settings?.allowLegacyExtensionDirectory = allowLegacyExtensionDirectory.state==NSControl.StateValue.on
    }

    override func enable(forCompiler compiler:String) -> Bool {
        return true
    }
}
